### PR TITLE
Support for mutil-value audience claims in JWT token

### PR DESCRIPTION
**Pull request by rstyp**
## Description

`requiredAudience` validation fails if `aud` claim contains multiple audiences in JWT token.

- Remove `jwtParser.requireAudience(config.getRequiredAudience());` to not to validate `aud` during `parseClaimsJws`.
- Add `validateAudience` method and call it from `createIdentity`

> Is this change a fix, improvement, new feature, refactoring, or other?

a fix

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

JWT Authentication

> How would you describe this change to a non-technical end user or system administrator?

Support for mutil-value audience claims in JWT token

## Related issues, pull requests, and links

#13442

## Documentation

( *) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# JWT Authentication
* Support for mutil-value audience claims in JWT token. ({issue}`13442 `)
```

### CODE REVIEW

**Review by lukasz-walkiewicz — COMMENTED** [review]
Looks like a good start but we need to add tests for this change.
Please take a look at `io.trino.server.security.TestResourceSecurity`. There are a few test for JWT authentication.

File: core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java [hunk 1]
```diff
@@ -26,32 +27,34 @@
...
 import static io.trino.server.security.jwt.JwtUtil.newJwtParserBuilder;
+import static java.lang.String.format;
 
 public class JwtAuthenticator
         extends AbstractBearerAuthenticator
 {
     private final JwtParser jwtParser;
     private final String principalField;
     private final UserMapping userMapping;
+    private final String requiredAudience;
```
[the remark this answers is not in the document]
Comment (rstyp) [reply core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java]: @lukasz-walkiewicz changed it to `Optional`

**Comment by lukasz-walkiewicz** [comment]
@rstyp Please sign CLA. See instructions from cla-bot.

**Comment by rstyp** [comment]
@lukasz-walkiewicz CLA is signed now.

File: core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java [hunk 2]
```diff
@@ -71,6 +75,32 @@ protected Optional<Identity> createIdentity(String token)
                 .build());
     }
 
+    private void validateAudience(Claims claims)
+    {
+        Object tokenAudience = claims.get(AUDIENCE);
```
Comment (lukasz-walkiewicz) [inline core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java:80]: change order, if `requiredAudience` is empty then getting audience from claims is redundant

File: core/trino-main/src/test/java/io/trino/server/security/TestResourceSecurity.java [hunk 3]
```diff
@@ -538,6 +552,34 @@ public void testJwtWithJwkAuthenticator()
         }
     }
 
+    @Test
+    public void testJwtAuthenticatorWithInvalidAudience()
```
Comment (lukasz-walkiewicz) [inline core/trino-main/src/test/java/io/trino/server/security/TestResourceSecurity.java:556]: I think we're missing a test with empty audience and without `http-server.authentication.jwt.required-audience` specified.

**Review by lukasz-walkiewicz — COMMENTED** [review]
Sorry for the delay @rstyp. Looks good, just a couple of minor comments.

**Review by lukasz-walkiewicz — APPROVED** [review]
Thank you @rstyp. LGTM :+1:
@kokosing could you please take a look as well and merge it?

**Comment by rstyp** [comment]
Thank you @lukasz-walkiewicz

**Comment by kokosing** [comment]
CI hit: https://github.com/trinodb/trino/issues/12818

**Comment by kokosing** [comment]
CI hit: https://github.com/trinodb/trino/issues/13288

**Comment by kokosing** [comment]
Merged, thanks
